iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/whatsapp/WhatsAppParser.java

```diff
@@ -793,6 +806,41 @@ private void parseWhatsAppAccount(InputStream is, ParseContext context, ContentH
         }
     }
 
+    private void fillAccountWithContactData(WAAccount account, IItemSearcher searcher, String dbPath) {
+        if (account == null || account.getWaName() != null) {
+            return;
+        }
+        try {
+            WAContactsDirectory contacts = getWAContactsDirectoryForPath(dbPath, searcher, ExtractorAndroidFactory.class);
+            WAContact c = contacts.getContact(account.getFullId());
+            if (c != null) {
+                account.setWaName(c.getDisplayName());
+                account.setStatus(c.getStatus());
+            }
+        } catch (Exception e) {
+           logger.warn("Error filling WhatsApp account with contact data: " + account.getFullId(), e);
+        }
+    }
+
+    private void fillAccountAvatar(WAAccount account, IItemSearcher searcher, String dbPath) {
+        if (account == null || account.getAvatar() != null || account.getAvatarPath() != null) {
+            return;
+        }
+        String filePath = Paths.get(dbPath).getParent().resolveSibling("files").resolve("me.jpg").toString().replace("\\", "/");
```

Comment:
This line is causing a critical (abort) error, if dbPath is a container (e.g. ZIP), because the path becomes something like "a.zip>>data/Root/data/com.whatsapp/databases/wa.db", which is not accepted by the `Path.get()` as it is not a valid path (contains ">").
Although wa.db is not usually inside a container, it is a possible situation.
I will modify the code, to avoid this exception.

Reply:
It can be inside a zip/tar FFS extraction when processed directly. It will be more common in the future when #2095 is merged.

Reply:
Great! I hadn't thought about that!